app/models/chargeback_rate_detail.rb

```diff
@@ -252,6 +252,18 @@ def self.default_rate_details_for(rate_type)
         end
 
         rate_details.push(detail_new)
+
+        if detail_new.chargeable_field.metric == 'derived_vm_allocated_disk_storage'
+          volume_types = CloudVolume.pluck(:volume_type).uniq.compact
```

Comment:
Perhaps `CloudVolume.where.not(:volume_type => nil).group(:volume_type).pluck(:volume_type).uniq` would be more efficient.